File: Tarea6_2.cpp
#include <iostream>
using namespace std;

int main() {
    // Arreglo de 6 números enteros
    int numeros[6] = {-8, 19, -3, 5, 13,100};

    // Ciclo for con limites correctos
    for (int i = 0; i < 6; i++) {

        //Mostrar posicion y valor
        cout << "Posicion [" << i << "] = " << numeros[i] << endl;
    }

    /*
      Si el ciclo supera el tamaño del arreglo
      (por ejemplo i <= 6 o i < 10),
      el programa intentaría acceder a posiciones
      de memoria que no pertenecen al arreglo.
      Esto puede provocar errores, resultados
      incorrectos o desbordamiento de memoria.
    */

    return 0;
}

File: Tarea6_2.java
public class Ejercicio2 {
    public static void main(String[] args) {

        // Arreglo con 6 números enteros
        int[] numeros = {-8, 19, -3, 5, 13, 100};

        // Ciclo for con límites correctos
        for (int i = 0; i <= numeros.length; i++) {

            // Mostrar posición y valor
            System.out.println("Posición " + i + " = " + numeros[i]);
        }

        /*
         * El arreglo tiene posiciones de 0 a 5.
         * Cuando i llega a 6, el programa intenta acceder
         * a una posición que no existe y genera el error:
         * ArrayIndexOutOfBoundsException
         */
    }
}

